runtime/lua/vim/_editor.lua

```diff
@@ -328,7 +328,10 @@ do
       -- message when there are zero dots.
       vim.api.nvim_command(('echo "%s"'):format(dots))
     end
+    startpos = startpos == nil and vim.fn.getpos("'[") or startpos
     if is_last_chunk then
+      vim.fn.setpos("'[", assert(startpos))
+      startpos = nil
       vim.api.nvim_command('redraw' .. (tick > 1 and '|echo ""' or ''))
     end
```

Comment:
This is what I meant:
```suggestion
    if is_last_chunk then
      if startpos then
        vim.fn.setpos("'[", assert(startpos))
        startpos = nil
      end
      vim.api.nvim_command('redraw' .. (tick > 1 and '|echo ""' or ''))
    else if startpos == nil then
      startpos = vim.fn.getpos("'[")
    end
```